examples/tests/sdk_tests/service/stateful.test.w

```diff
@@ -0,0 +1,50 @@
+bring cloud;
+bring util;
+bring http;
+
+// hack: only supported in the "sim" target for now
+if util.env("WING_TARGET") == "sim" {
+  class MyService {
+    b: cloud.Bucket;
+    body: str;
+
+    pub s: cloud.Service;
+
+    init(body: str) {
+      this.b = new cloud.Bucket();
+      this.body = body;
+
+      // it's idiomatic to just pass `this` here and implement the callbacks on the current object.
+      this.s = new cloud.Service(inflight () => {
+        log("starting service");
+        util.sleep(1s);
+        this.b.put("ready", "true");
+        let state = 456;
+  
+        return () => {
+          log("stopping service");
+          log("state is: ${state}");
+    
+          // make sure inflight state is presistent across onStart/onStop
+          assert(state == 456);
+        };
+      });
+    }
+
+    pub inflight access() {
+      // when access() is called we expect the service to have completed initialization
+      this.b.get("ready");
```

Comment:
How does this work in the cloud / real world?

For example, if I do:

```js
let s = new MyService();
new cloud.Function(inflight () => {
  s.access();
});
```

How are we guaranteeing that this AWS Lambda won't be called until the service is started (or that `access()` doesn't run until the service has started)? Would the Terraform deployment have to block until the all `cloud.Service`'s `onStart` functions have started? Keep in mind the inflight method `access()` doesn't require permissions to `cloud.Service` since it doesn't use `this.s`.

Reply:
yes, we would need the deployment of the `cloud.Service` to only finish when the handler returns. In CloudFormation, for example, there's a way to [signal to the engine](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-signal.html) when a resource has completed initialization. I am certain there's a way to do that in Terraform as well.